service/history/shardContext.go

```diff
@@ -562,11 +568,11 @@ func (s *shardContextImpl) AppendHistoryEvents(request *persistence.AppendHistor
 		if _, ok := err0.(*persistence.ConditionFailedError); ok {
 			// Inserting a new event failed, lets try to overwrite the tail
 			request.Overwrite = true
-			return s.historyMgr.AppendHistoryEvents(request)
+			return size, s.historyMgr.AppendHistoryEvents(request)
 		}
 	}
 
-	return err0
+	return 0, err0
```

Comment:
Why return `0` not `size` when no error?

Reply:
Good catch.  Fixed.